ghost/member-attribution/lib/attribution.js

```diff
@@ -1,10 +1,71 @@
 /**
- * @typedef {object} Attribution
- * @prop {string|null} [id]
- * @prop {string|null} [url]
- * @prop {string} [type]
+ * @typedef {object} AttributionResource
+ * @prop {string|null} id
+ * @prop {string|null} url
+ * @prop {'page'|'post'|'author'|'tag'|'url'} type
+ * @prop {string|null} title
  */
 
+class Attribution {
+    #urlTranslator;
+
+    /**
+     * @param {object} data
+     * @param {string|null} [data.id]
+     * @param {string|null} [data.url]
+     * @param {'page'|'post'|'author'|'tag'|'url'} [data.type]
+     */
+    constructor({id, url, type}, {urlTranslator}) {
+        /** @type {string|null} */
```

Comment:
We shouldn't need these as the types can be inferred by the `@param` tags above - no harm in having them tho!

Reply:
Fixed in https://github.com/TryGhost/Ghost/commit/02168b41ceb5b75005584b791b80d5e6beaff11a